File: win-prog/ProgWinCS/Brushes and Pens/TextureBrushDemo/TextureBrushDemo.cs
//-----------------------------------------------
// TextureBrushDemo.cs © 2001 by Charles Petzold
//-----------------------------------------------
using System;
using System.Drawing;
using System.Drawing.Drawing2D;
using System.Windows.Forms;

class TextureBrushDemo: PrintableForm
{
     MenuItem     miChecked;
     TextureBrush tbrush;

     public new static void Main()
     {
          Application.Run(new TextureBrushDemo());
     } 
     public TextureBrushDemo()
     {
          Text = "Texture Brush Demo";

          Image image = Image.FromFile(
               "..\\..\\..\\..\\Images and Bitmaps\\Apollo11FullColor.jpg");

          tbrush = new TextureBrush(image, new Rectangle(95, 0, 50, 55));

          Menu = new MainMenu();
          Menu.MenuItems.Add("&Wrap-Mode");

          foreach (WrapMode wm in Enum.GetValues(typeof(WrapMode)))
          {
               MenuItem mi = new MenuItem();
               mi.Text     = wm.ToString(); 
               mi.Click   += new EventHandler(MenuWrapModeOnClick);
               Menu.MenuItems[0].MenuItems.Add(mi);
          }
          miChecked = Menu.MenuItems[0].MenuItems[0];
          miChecked.Checked = true;
     }
     void MenuWrapModeOnClick(object obj, EventArgs ea)
     {
          miChecked.Checked = false;
          miChecked = (MenuItem) obj;
          miChecked.Checked = true;

          tbrush.WrapMode = (WrapMode)miChecked.Index;
          Invalidate();
     }
     protected override void DoPage(Graphics grfx, Color clr, int cx, int cy)
     {
          grfx.FillEllipse(tbrush, 0, 0, 2 * cx / 3, 2 * cy / 3);
          grfx.FillEllipse(tbrush, cx / 3, cy / 3, 2 * cx / 3, 2 * cy / 3);
     }
}

File: win-prog/ProgWinVB/Brushes and Pens/TextureBrushDemo/TextureBrushDemo.vb
'-------------------------------------------------
' TextureBrushDemo.vb (c) 2002 by Charles Petzold
'-------------------------------------------------
Imports System
Imports System.Drawing
Imports System.Drawing.Drawing2D
Imports System.Windows.Forms

Class TextureBrushDemo
    Inherits PrintableForm

    Private miChecked As MenuItem
    Private tbr As TextureBrush

    Shared Shadows Sub Main()
        Application.Run(New TextureBrushDemo())
    End Sub

    Sub New()
        Text = "Texture Brush Demo"

        Dim img As Image = Image.FromFile( _
                "..\..\..\Images and Bitmaps\Apollo11FullColor.jpg")
        tbr = New TextureBrush(img, New Rectangle(95, 0, 50, 55))

        Menu = New MainMenu()
        Menu.MenuItems.Add("&Wrap-Mode")

        Dim wm As WrapMode
        For Each wm In System.Enum.GetValues(GetType(WrapMode))
            Dim mi As New MenuItem()
            mi.Text = wm.ToString()
            AddHandler mi.Click, AddressOf MenuWrapModeOnClick
            Menu.MenuItems(0).MenuItems.Add(mi)
        Next wm

        miChecked = Menu.MenuItems(0).MenuItems(0)
        miChecked.Checked = True
    End Sub

    Sub MenuWrapModeOnClick(ByVal obj As Object, ByVal ea As EventArgs)
        miChecked.Checked = False
        miChecked = DirectCast(obj, MenuItem)
        miChecked.Checked = True

        tbr.WrapMode = CType(miChecked.Index, WrapMode)
        Invalidate()
    End Sub

    Protected Overrides Sub DoPage(ByVal grfx As Graphics, _
            ByVal clr As Color, ByVal cx As Integer, ByVal cy As Integer)
        grfx.FillEllipse(tbr, 0, 0, 2 * cx \ 3, 2 * cy \ 3)
        grfx.FillEllipse(tbr, cx \ 3, cy \ 3, 2 * cx \ 3, 2 * cy \ 3)
    End Sub
End Class
